Another Keyword Protection error

**Describe the bug**
It seems like Tower of the Magistrate has the same issue that Pippin had as far as protection goes, it soft locks the game.

`Game-0 > java.lang.RuntimeException: Error in Keyword Protection from artifacts for card Falkenrath Gorger
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:128)
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:86)
	at forge.game.card.Card.getKeywordForStaticAbility(Card.java:4873)
	at forge.game.card.Card.addChangedCardKeywords(Card.java:4850)
	at forge.game.ability.effects.ProtectEffect.resolve(ProtectEffect.java:156)
	at forge.game.ability.AbilityApiBased.resolve(AbilityApiBased.java:41)
	at forge.game.ability.AbilityUtils.resolveApiAbility(AbilityUtils.java:1462)
	at forge.game.ability.AbilityUtils.resolve(AbilityUtils.java:1430)
	at forge.game.zone.MagicStack.resolveStack(MagicStack.java:553)
	at forge.game.phase.PhaseHandler.startFirstTurn(PhaseHandler.java:1154)
	at forge.game.GameAction.startGame(GameAction.java:2113)
	at forge.game.Match.startGame(Match.java:90)
	at forge.gamemodes.match.HostedMatch$2.run(HostedMatch.java:255)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.RuntimeException: unknown protection keyword: Protection from artifacts
	at forge.game.card.CardFactoryUtil.getProtectionValid(CardFactoryUtil.java:683)
	at forge.game.card.CardFactoryUtil.addReplacementEffect(CardFactoryUtil.java:2567)
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:116)
	... 15 more`

**Forge Version:** forge-gui-desktop-1.6.60-SNAPSHOT-01-15

--- a/forge-ai/src/main/java/forge/ai/ability/ProtectAi.java
+++ b/forge-ai/src/main/java/forge/ai/ability/ProtectAi.java
@@ -72,8 +72,8 @@ public class ProtectAi extends SpellAbilityAi {
             return null;
         }
         final List<String> choices = ProtectEffect.getProtectionList(sa);
-        if (threat.isArtifact() && choices.contains("artifacts")) {
-            return "artifacts";
+        if (threat.isArtifact() && choices.contains("Artifact")) {
+            return "Artifact";
         }
         if (threat.isBlack() && choices.contains("black")) {
             return "black";
@@ -157,7 +157,7 @@ public class ProtectAi extends SpellAbilityAi {
             }
         });
         return list;
-    } // getProtectCreatures()
+    }
 
     @Override
     protected boolean checkPhaseRestrictions(final Player ai, final SpellAbility sa, final PhaseHandler ph) {

--- a/forge-game/src/main/java/forge/game/ability/effects/ProtectAllEffect.java
+++ b/forge-game/src/main/java/forge/game/ability/effects/ProtectAllEffect.java
@@ -125,7 +125,6 @@ public class ProtectAllEffect extends SpellAbilityEffect {
                 }
             }
         }
-
-    } // protectAllResolve()
+    }
 
 }

--- a/forge-game/src/main/java/forge/game/ability/effects/ProtectEffect.java
+++ b/forge-game/src/main/java/forge/game/ability/effects/ProtectEffect.java
@@ -133,7 +133,7 @@ public class ProtectEffect extends SpellAbilityEffect {
 
         List<String> gainsKWList = Lists.newArrayList();
         for (String type : gains) {
-            if (isChoice && sa.getParam("Choices").equals("CardType")) {
+            if (CardType.isACardType(type)) {
                 gainsKWList.add("Protection:" + type);
             }  else {
                 gainsKWList.add(TextUtil.concatWithSpace("Protection from", type));

--- a/forge-gui/res/cardsfolder/a/apostles_blessing.txt
+++ b/forge-gui/res/cardsfolder/a/apostles_blessing.txt
@@ -1,5 +1,5 @@
 Name:Apostle's Blessing
 ManaCost:1 WP
 Types:Instant
-A:SP$ Protection | Cost$ 1 WP | ValidTgts$ Creature.YouCtrl,Artifact.YouCtrl | TgtPrompt$ Select target artifact or creature you control | Gains$ Choice | Choices$ AnyColor,artifacts | SpellDescription$ Target artifact or creature you control gains protection from artifacts or from the color of your choice until end of turn.
+A:SP$ Protection | ValidTgts$ Creature.YouCtrl,Artifact.YouCtrl | TgtPrompt$ Select target artifact or creature you control | Gains$ Choice | Choices$ AnyColor,Artifact | SpellDescription$ Target artifact or creature you control gains protection from artifacts or from the color of your choice until end of turn.
 Oracle:({W/P} can be paid with either {W} or 2 life.)\nTarget artifact or creature you control gains protection from artifacts or from the color of your choice until end of turn.

--- a/forge-gui/res/cardsfolder/j/jareth_leonine_titan.txt
+++ b/forge-gui/res/cardsfolder/j/jareth_leonine_titan.txt
@@ -3,6 +3,6 @@ ManaCost:3 W W W
 Types:Legendary Creature Cat Giant
 PT:4/7
 T:Mode$ Blocks | ValidCard$ Card.Self | Execute$ TrigPump | TriggerDescription$ Whenever CARDNAME blocks, it gets +7/+7 until end of turn.
-A:AB$ Protection | Cost$ W | Gains$ Choice | Choices$ AnyColor | SpellDescription$ CARDNAME gains protection from the color of your choice until end of turn.
+A:AB$ Protection | Cost$ W | Gains$ Choice | Choices$ AnyColor | SpellDescription$ NICKNAME gains protection from the color of your choice until end of turn.
 SVar:TrigPump:DB$ Pump | Defined$ Self | NumAtt$ 7 | NumDef$ 7
 Oracle:Whenever Jareth, Leonine Titan blocks, it gets +7/+7 until end of turn.\n{W}: Jareth gains protection from the color of your choice until end of turn.

--- a/forge-gui/res/cardsfolder/j/jeweled_spirit.txt
+++ b/forge-gui/res/cardsfolder/j/jeweled_spirit.txt
@@ -3,6 +3,6 @@ ManaCost:3 W W
 Types:Creature Spirit
 PT:3/3
 K:Flying
-A:AB$ Protection | Cost$ Sac<2/Land> | Gains$ Choice | Choices$ AnyColor,artifacts | SpellDescription$ CARDNAME gains protection from artifacts or from the color of your choice until end of turn.
+A:AB$ Protection | Cost$ Sac<2/Land> | Gains$ Choice | Choices$ AnyColor,Artifact | SpellDescription$ CARDNAME gains protection from artifacts or from the color of your choice until end of turn.
 AI:RemoveDeck:All
 Oracle:Flying\nSacrifice two lands: Jeweled Spirit gains protection from artifacts or from the color of your choice until end of turn.

--- a/forge-gui/res/cardsfolder/r/razor_barrier.txt
+++ b/forge-gui/res/cardsfolder/r/razor_barrier.txt
@@ -1,5 +1,5 @@
 Name:Razor Barrier
 ManaCost:1 W
 Types:Instant
-A:SP$ Protection | Cost$ 1 W | ValidTgts$ Permanent.YouCtrl | TgtPrompt$ Select target permanent you control | Gains$ Choice | Choices$ AnyColor,artifacts | SpellDescription$ Target permanent you control gains protection from artifacts or from the color of your choice until end of turn.
+A:SP$ Protection | ValidTgts$ Permanent.YouCtrl | TgtPrompt$ Select target permanent you control | Gains$ Choice | Choices$ AnyColor,Artifact | SpellDescription$ Target permanent you control gains protection from artifacts or from the color of your choice until end of turn.
 Oracle:Target permanent you control gains protection from artifacts or from the color of your choice until end of turn.

--- a/forge-gui/res/cardsfolder/t/tel_jilad_defiance.txt
+++ b/forge-gui/res/cardsfolder/t/tel_jilad_defiance.txt
@@ -1,6 +1,6 @@
 Name:Tel-Jilad Defiance
 ManaCost:1 G
 Types:Instant
-A:SP$ Protection | Cost$ 1 G | ValidTgts$ Creature | TgtPrompt$ Select target creature | Gains$ artifacts | SpellDescription$ Target creature gains protection from artifacts until end of turn. | SubAbility$ DBDraw
+A:SP$ Protection | ValidTgts$ Creature | Gains$ Artifact | SpellDescription$ Target creature gains protection from artifacts until end of turn. | SubAbility$ DBDraw
 SVar:DBDraw:DB$ Draw | NumCards$ 1 | SpellDescription$ Draw a card.
 Oracle:Target creature gains protection from artifacts until end of turn.\nDraw a card.

--- a/forge-gui/res/cardsfolder/t/tower_of_the_magistrate.txt
+++ b/forge-gui/res/cardsfolder/t/tower_of_the_magistrate.txt
@@ -2,5 +2,5 @@ Name:Tower of the Magistrate
 ManaCost:no cost
 Types:Land
 A:AB$ Mana | Cost$ T | Produced$ C | SpellDescription$ Add {C}.
-A:AB$ Protection | Cost$ 1 T | ValidTgts$ Creature | TgtPrompt$ Select target creature | Gains$ artifacts | SpellDescription$ Target creature gains protection from artifacts until end of turn.
+A:AB$ Protection | Cost$ 1 T | ValidTgts$ Creature | TgtPrompt$ Select target creature | Gains$ Artifact | SpellDescription$ Target creature gains protection from artifacts until end of turn.
 Oracle:{T}: Add {C}.\n{1}, {T}: Target creature gains protection from artifacts until end of turn.

